lib/ios/format.js

```diff
@@ -0,0 +1,225 @@
+${uiColorBlock}
+}
+`;
+  },
+});
+
+// Register tokens-swift-format for iOS light token generation
+StyleDictionary.registerFormat({
+  name: "tokens-swift-format",
+  formatter: function ({ dictionary, file, options }) {
+    console.log(
+      `Inside tokens-swift-format formatter for ${file.destination}`
+    );
+
+    const brandName = file.destination
+      .split("/")
+      .pop()
+      .replace("Tokens.swift", ""); // Extract brand name from the file path
+
+    // Generate SwiftUI light colors
+    const swiftUIColorBlock = dictionary.allProperties
+      .filter((token) => !token.path[0].startsWith("color")) // Exclude primitive colors, only keep semantic tokens
+      .map((token) => {
+        let name = toCamelCase(token.path.slice(2).join("-")).replace(
+          "Default",
+          ""
+        ); // Convert to camelCase, removing first two parts
+        let value = transformValue(token.original.value);
+
+        return `    public var ${name}: Color { ${brandName.replace(
+          /Dark|Light/g,
+          ""
+        )}${value} }`;
+      })
+      .join("\n");
+
+    // Generate UIKit light colors
+    const uiColorBlock = dictionary.allProperties
+      .filter((token) => !token.path[0].startsWith("color")) // Exclude primitive colors, only keep semantic tokens
+      .map((token) => {
+        let name = toCamelCase(token.path.slice(2).join("-")).replace(
+          "Default",
+          ""
+        ); // Convert to camelCase, removing first two parts
+        let value = transformValue(token.original.value);
+
+        return `    public var ${name}: UIColor { ${brandName.replace(
+          /Dark|Light/g,
+          ""
+        )}UI${value} }`;
+      })
+      .join("\n");
+
+    return `import SwiftUI
+  
+  // Generated on ${getGeneratedDate()} by https://github.com/warp-ds/tokens
+  struct ${brandName}TokenProvider: TokenProvider {
+  ${swiftUIColorBlock}
+  }
+  
+  struct ${brandName}UITokenProvider: UITokenProvider {
+  ${uiColorBlock}
+  }`;
+  },
+});
+
+// Register colors-swift-format for iOS light color generation
+StyleDictionary.registerFormat({
+  name: "colors-swift-format",
+  formatter: function ({ dictionary, file, options }) {
+    console.log(
+      `Inside colors-swift-format formatter for ${file.destination}`
+    );
+
+    const brandName = file.destination
+      .split("/")
+      .pop()
+      .replace("Colors.swift", ""); // Extract brand name from the file path
+
+    // Generate SwiftUI colors
+    const swiftUIColorBlock = dictionary.allProperties
+      .filter((token) => !token.path[0].startsWith("color")) // Exclude primitive colors, only keep component colors
+      .filter((token) => !token.path[1].startsWith("color")) // Exclude semantic tokens, only keep component colors
+      .map((token) => {
+        let name = toCamelCase(token.path.slice(1).join("-")).replace(
+          "Default",
+          ""
+        ); // Convert to camelCase, removing first two parts
```

Comment:
I guess it's the first part that's removed here (`slice(1)`)?

Reply:
nice catch 👍 there were two of these, changed both